pkg/controller/clusterpool/clusterpool_controller_test.go

```diff
@@ -166,6 +189,198 @@ func TestReconcileClusterPool(t *testing.T) {
+			expectError:                  true,
+		},
+		{
+			name: "cp with inventory and cdc patch broken is not valid - BrokenBySyntax",
+			existing: []runtime.Object{
+				inventoryPoolBuilder.Build(testcp.WithSize(1)),
+				testcdc.FullBuilder(testNamespace, "test-cdc-1", scheme).Build(
+					testcdc.WithPatch("/broken/path", "replace", "x"),
+				),
+			},
+			inventory:                    true,
+			expectedTotalClusters:        0,
+			expectedObservedSize:         0,
+			expectedObservedReady:        0,
+			expectedInventoryVaildStatus: corev1.ConditionFalse,
+			expectedInventoryMessage:     map[string][]string{"BrokenBySyntax": {"test-cdc-1"}},
+			expectedPoolVersion:          "06983eaafac7f695",
+			expectedCDCurrentStatus:      corev1.ConditionUnknown,
+			expectError:                  true,
+		},
+		{
+			name: "cp with inventory and cd provisioning failed is not valid - BrokenByCloud",
+			existing: []runtime.Object{
+				inventoryPoolBuilder.Build(testcp.WithSize(1)),
+				testcdc.FullBuilder(testNamespace, "test-cdc-1", scheme).Build(),
+				testcd.FullBuilder("c1", "c1", scheme).Build(
+					testcd.WithPoolVersion("06983eaafac7f695"),
+					testcd.WithUnclaimedClusterPoolReference(testNamespace, testLeasePoolName),
+					testcd.WithCustomization("test-cdc-1"),
+					testcd.Broken(),
+				),
+			},
+			inventory:                    true,
+			expectedTotalClusters:        0,
+			expectedObservedSize:         1,
+			expectedObservedReady:        0,
+			expectedInventoryVaildStatus: corev1.ConditionFalse,
+			expectedInventoryMessage:     map[string][]string{"BrokenByCloud": {"test-cdc-1"}},
+			expectedPoolVersion:          "06983eaafac7f695",
+			expectedAssignedCDCs:         1,
+		},
+		{
+			name: "cp with inventory and good cdc is valid, cd created",
+			existing: []runtime.Object{
+				inventoryPoolBuilder.Build(testcp.WithSize(1)),
+				testcdc.FullBuilder(testNamespace, "test-cdc-1", scheme).Build(),
+				unclaimedCDBuilder("c1").Build(
+					testcd.WithCustomization("test-cdc-1"),
+					testcd.Running(),
+				),
+			},
+			inventory:                    true,
+			expectedTotalClusters:        0,
+			expectedObservedSize:         1,
+			expectedObservedReady:        1,
+			expectedInventoryVaildStatus: corev1.ConditionTrue,
+			expectedPoolVersion:          "06983eaafac7f695",
+			expectedAssignedCDCs:         1,
+		},
+		{
+			// Second prioritize by last used
+			name: "cp with inventory - correct prioritization - same status",
+			existing: []runtime.Object{
+				initializedPoolBuilder.Build(
+					testcp.WithSize(1),
+					testcp.WithInventory([]string{"test-cdc-successful-old", "test-cdc-unused-new"}),
+				),
+				testcdc.FullBuilder(testNamespace, "test-cdc-successful-old", scheme).Build(
+					testcdc.WithLastStatus(hivev1.LastApplySucceeded),
+					testcdc.WithLastTime(nowish.Add(-time.Hour)),
+				),
+				testcdc.FullBuilder(testNamespace, "test-cdc-unused-new", scheme).Build(
+					testcdc.WithLastTime(nowish),
```

Comment:
Given the test name, did you mean to set the status the same here? To prove that the older one comes out first?
```suggestion
					testcdc.WithLastStatus(hivev1.LastApplySucceeded),
					testcdc.WithLastTime(nowish),
```